test/mnist-pytorch/load_dataset.py

```diff
@@ -0,0 +1,11 @@
+import torchvision
+
+if __name__ == '__main__':
+
+    for trainset in [True, False]:
```

Comment:
Hard to understand what is going on here? Is it downloading training data in one case and test in the other? In that case, I suggest splitting into two explicit stages.